src/test/java/uk/gov/digital/ho/proving/income/validator/FrequencyCalculatorTest.java

```diff
@@ -504,15 +507,15 @@ private long randomBetween(int min, int max) {
         return new Random().nextInt(max - min + 1) + min;
     }
 
-    private void verifyLogMessage(final String message) {
+    private void verifyLogMessage(final String message, LogEvent event) {
         ArgumentCaptor<ILoggingEvent> captor = ArgumentCaptor.forClass(ILoggingEvent.class);
         verify(mockAppender, times(2)).doAppend(captor.capture());
 
         List<ILoggingEvent> loggingEvents = captor.getAllValues();
         for (ILoggingEvent loggingEvent : loggingEvents) {
             LoggingEvent logEvent = (LoggingEvent) loggingEvent;
             if (logEvent.getMessage().equals(message) && logEvent.getLevel().equals(Level.INFO) &&
-                ((ObjectAppendingMarker)logEvent.getArgumentArray()[0]).getFieldName().equals("event_id")) {
+                loggingEvent.getArgumentArray()[0].equals(new ObjectAppendingMarker("event_id", event))) {
```

Comment:
This is the correct way to check the LogEvent but you'll need to make other tests work this way.